uc/uc.go

```diff
@@ -61,34 +55,36 @@ func New(client Client, enableDebug bool) (*calConn, error) {
 	c := &calConn{}
 	c.debug = enableDebug
 	c.client = client
-	c.clientOpts = c.client.GetClientOptions()
+	c.clientOpts, retErr = c.client.GetClientOptions()
+	if retErr != nil {
+		return nil, fmt.Errorf("New: Error getting client options: %w", retErr)
+	}
 	c.transferCount = 0
 	c.okStr = "6[0,{}]"
+	c.ucdb = &UncagedDB{}
+	bookList, retErr := c.client.GetDeviceBookList()
 	if retErr != nil {
-		return nil, retErr
+		return nil, fmt.Errorf("New: Error getting booklist from device: %w", retErr)
+	}
+	c.ucdb.initDB(bookList)
+	if c.deviceInfo, retErr = c.client.GetDeviceInfo(); retErr != nil {
+		return nil, fmt.Errorf("New: Error getting info from device: %w", retErr)
 	}
-	c.ucdb = &UncagedDB{}
-	c.ucdb.initDB(c.client.GetDeviceBookList())
-	c.deviceInfo = c.client.GetDeviceInfo()
-	udpReply := make(chan string)
 	// Calibre listens for a 'hello' UDP packet on the following
 	// five ports. We try all five ports concurrently
 	c.client.UpdateStatus(SearchingCalibre, -1)
 	bcastPorts := []int{54982, 48123, 39001, 44044, 59678}
+	wg := &sync.WaitGroup{}
+	mu := &sync.Mutex{}
 	for _, p := range bcastPorts {
-		go c.findCalibre(p, udpReply)
+		wg.Add(1)
+		go c.findCalibre(p, mu, wg)
 	}
```

Comment:
This should be like this to follow the usual convention:

```go
	var wg sync.WaitGroup
	var mu sync.Mutex
	for _, p := range bcastPorts {
		wg.Add(1)
		go c.findCalibre(p, &mu, &wg)
	}
```

Reply:
Took me a few moments to figure out what you had actually suggested, but yeah, have made that change.